### ISSUE TITLE
Can you pause the learning process and extract the model for some testing?

### ISSUE THREAD

**Issue by Kingboom4** [Russell91/TensorBox#68]
Hi guys, I do apologise if this is a stupid question but I was wondering whether I could pause the learning process and run` evaluate.py` or` evaluate.ipynb`, I can't seem to be able to find the model in the output directory, am I missing something?

**Comment by Russell91** [comment]
Yes, good question. The model will indeed be in whatever --logdir you specified, as a *.ckpt file. At any time, you can use a second GPU to run the evaluate.ipynb on the most recent ckpt. You can also pause the script with ctrl-z, but unfortunately this will only free the GPUs processing, not it's memory. So your best bet is to either get a second GPU, look up how to tell tensorflow to reserve memory, or run the evaluate file on the CPU.

**Comment by Kingboom4** [comment]
@Russell91 Awesome! Will try reserve some memory for testing, if I can't do it will just run it on the CPU and see how long it takes. Thanks again.

**Comment by Kingboom4** [comment]
@Russell91 Sorry to bother you again, I was just reading the `evaluate.py` and `evaluate.ipynb` and noticed that they take in `*.idl` as parameters for the testing set, I am wondering whether there is a script available to convert between `*.json` and `*.idl`.

**Comment by botasb** [comment]
Kingboom4. I just created myself a small script to convert my json files in idl. You should be able if you know some coding (I guess you know.). BTW, pay attention to the evaluate.py and the ipynb files; there are hard encoded paths to the examples dataset/files (brainwash) and checkpoint file. You would need to change it to your files/paths.

**Comment by Kingboom4** [comment]
Thanks for the reply @botasb would you mind sharing the script that you made for converting from `*.json` to `*.idl`, I do apologise if this an unreasonable request - would just like to see how exactly you did it.

**Comment by Russell91** [comment]
The parse function works on both json and idl, so you can just change the
file path and it will work.

**Comment by Kingboom4** [comment]
@Russell91 I see, didn't know that - will just do that instead, thanks Russell.

**Comment by Kingboom4** [comment]
@Russell91 Everything is working flawlessly with the `evaluate.ipynb` script apart from the last section, I am not sure what I am doing wrong.

Here is a snippet of the code and error.

![image](https://cloud.githubusercontent.com/assets/5938514/18404042/1da13c16-76e0-11e6-9939-81f0ad838549.png)

**Comment by Kingboom4** [comment]
After adding a try and catch block to see what exactly the error is, I get the following: 

`RuntimeError: command './utils/annolist/doRPC.py --minOverlap 0.500000 /home/waj/tensorbox/annotations_testing.json /media/waj/Hitachi Storage/Deep learning project storage/Tensorflow_output/7350000_val_hypes.idl' return with error (code 1): Minimum width: 0 height: 0`

Any suggestions?

**Comment by Kingboom4** [comment]
Just ended up writing the commands manually in the terminal and executing it one by one - works great, would have been nice to have it working on the evaluate script but not sure why it doesn't like it.

**Comment by Kingboom4** [comment]
I think I've figured out what was wrong, the `*.ipynb` script is happy to accept paths without the backwards slash `\` used for spaces in the terminal but when the `pred_idl` is used for passing the command it requires the backwards slashes to be there.

In other words instead of this:

`/media/waj/Hitachi Storage/Deep learning project storage/Tensorflow_output/7350000_val_hypes.idl`

it needs to be:
`/media/waj/Hitachi\ Storage/Deep\ learning\ project\ storage/Tensorflow_output/`

**Comment by usamaoDATA** [comment]
i pressed ctr+z after about 2500 steps while training on my own data set of about 25 images. but i see no *.ckpt file in the output dir, i see images with actual boxes i labelled and the predicted boxes (lots of red and green rectangles on these images). a hypes.json file and a file called "events.out.tfevents.1477297029.oDATA-MBP"

oDATA-MBP is my laptops name

how do i get the *.ckpt file

**Comment by rubysolo** [comment]
@usamaoDATA Did you use the included `overfeat.json` hype file?  It has the `save_iter` variable set to 10000, so you'll need to run that many iterations before a checkpoint is created or reduce that number to something lower.
